cudamapper/src/index.cu

```diff
@@ -40,3 +40,135 @@ std::unique_ptr<Index> Index::create_index(std::shared_ptr<DeviceAllocator> allo
 
 } // namespace cudamapper
 } // namespace claragenomics
+
+
+
+namespace claragenomics
+{
+namespace cudamapper
+{
+
+    IndexCache::IndexCache(const Index& index,
+                           const read_id_t first_read_id_in,
+                           const std::uint64_t kmer_size_in,
+                           const std::uint64_t window_size_in)
+            : first_read_id(first_read_id_in),
+              kmer_size(kmer_size_in),
+              window_size(window_size_in)
+    {
+        CGA_NVTX_RANGE(profiler, "cache_index");
+        {
+            auto const & src = index.representations();
+            auto const sz = src.size();
+            representations_.resize(sz);
+            cudautils::device_copy_n(src.data(), sz, representations_.data());
```

Comment:
This is a personal preference, I find
```
representations_.resize(index.representations().size());
cudautils::device_copy_n(index.representations().data(), index.representations().size(), representations_.data());
```
to be more readable, especially as in that case there is no need for additional scopes (but again it's a personal preference)

Reply:
the reason is I keep using the same name auto const& src to represent different data types, therefore I used extra scope.

Reply:
I agree, but doing it in two lines like I suggested does not require you to explicitly specify the type and if anyone is unsure of what's input and what's output they can check the signature (like with any other function)

Reply:
this was a minor case of conflict between personal preferences. I changed to your suggested way. The reason I liked the other way was the code looked more concise and easier to read.